### ISSUE TITLE
CNN-RNN: A Unified Framework for Multi-label Image Classification

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/deep-learning-readinglist#62]
https://www.ics.uci.edu/~yyang8/research/cnn-rnn/cnn-rnn-cvpr2016.pdf

**Comment by sweaterr** [comment]
#### CNN-RNN: A Unified Framework for Multi-label Image Classification

In this paper, we propose a unified CNN-RNN framework for multi-label image classification, which effectively learns both the semantic redundancy and the co-occurrence dependency in an end-to-end way.

![image](https://cloud.githubusercontent.com/assets/1518919/26393762/cc033734-40a5-11e7-99f2-5a0d8c0058b7.png)

빨간점: 레이블 
파란점: 이미지 임베딩
검은점: 파란색 점인 이미지가 거기 맞는 레이블인 빨간점 쪽으로 이동 

Compared with state-of-the-art multi-label image classification methods, the proposed RNN framework has several advantages:

* end-to-end model
* more compact and more powerful model of high-order label co-occurrence dependency
* implicit attention mechanism

We evaluate the proposed CNN-RNN framework with exhaustive experiments on public multi-label benchmark datasets inlcuding NUS-WIDE, Microsoft COCO, and PASCAL VOC 2007.

3.2. Model

We propose a novel CNN-RNN framework for multilabel classification problem

![image](https://cloud.githubusercontent.com/assets/1518919/26394632/81228334-40a8-11e7-9900-e12a734278e0.png)

**Comment by sweaterr** [comment]
![image](https://cloud.githubusercontent.com/assets/1518919/26395970/b370386e-40ac-11e7-9385-562eaadc4b7c.png)

**Comment by sweaterr** [comment]
we employ the beam search algorithm to find the top-ranked prediction path

**Comment by sweaterr** [comment]
3.4. Training
Learning CNN-RNN models can be achieved by using the cross-entropy loss on the softmax normalization of score softmax(s(t)) and employing back-propagation through time algorithm.

**Comment by sweaterr** [comment]
Evaluation

The precision and recall of the generated labels are employed as evaluation metrics.
